Jellyfin.Api/Controllers/MediaSegmentController.cs

```diff
@@ -0,0 +1,150 @@
+    /// <param name="type">Segment type.</param>
+    /// <param name="typeIndex">Optional: If you want to add a type multiple times to the same itemId increment it.</param>
+    /// <param name="action">Optional: Creator recommends an action.</param>
+    /// <param name="comment">Optional: A comment.</param>
+    /// <response code="200">Segments returned.</response>
+    /// <response code="400">Missing query parameter.</response>
+    /// <returns>An <see cref="OkResult"/>containing the queryresult of segment.</returns>
+    [HttpPost("Segment")]
+    [Authorize(Policy = Policies.RequiresElevation)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<QueryResult<MediaSegment>>> PostSegment(
+        [FromQuery, Required] long startTicks,
+        [FromQuery, Required] long endTicks,
+        [FromQuery, Required] Guid itemId,
+        [FromQuery, Required] int streamIndex,
+        [FromQuery, Required] MediaSegmentType type,
+        [FromQuery, DefaultValue(0)] int typeIndex,
+        [FromQuery, DefaultValue(MediaSegmentAction.Auto)] MediaSegmentAction action,
+        [FromQuery] string comment)
+    {
+        var newMediaSegment = new MediaSegment()
+        {
+            StartTicks = startTicks,
+            EndTicks = endTicks,
+            ItemId = itemId,
+            StreamIndex = streamIndex,
+            Type = type,
+            TypeIndex = typeIndex,
+            Action = action,
+            Comment = comment
+        };
+
+        var segment = await _mediaSegmentManager.CreateMediaSegmentAsync(newMediaSegment).ConfigureAwait(false);
+
+        return new QueryResult<MediaSegment>(
+            new List<MediaSegment>
+            {
+                segment
+            });
+    }
+
+    /// <summary>
+    /// Create or update multiple media segments. See /MediaSegment/Segment for required properties.
+    /// </summary>
+    /// <param name="segments">All segments that should be added.</param>
+    /// <response code="200">Segments returned.</response>
+    /// <response code="400">Invalid segments.</response>
+    /// <returns>An <see cref="OkResult"/>containing the queryresult of segment.</returns>
+    [HttpPost]
+    [Authorize(Policy = Policies.RequiresElevation)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<QueryResult<MediaSegment>>> PostSegments(
+        [FromBody, Required] IEnumerable<MediaSegment> segments)
+    {
+        var nsegments = await _mediaSegmentManager.CreateMediaSegmentsAsync(segments).ConfigureAwait(false);
+
+        return new QueryResult<MediaSegment>(nsegments.ToList());
+    }
+
+    /// <summary>
+    /// Delete media segments. All query parameters can be freely defined.
+    /// </summary>
+    /// <param name="itemId">Optional: All segments with MediaSourceId.</param>
```

Comment:
Parameter is named itemId but the description tells me it is the mediaSourceId?

Reply:
I noticed that all id's are called 'itemId' while the api calls these streams a mediaSourceId. So it's just a hint which id you need from the api. Might be confusing